src/lib.js

```diff
@@ -193,31 +193,44 @@ var curlConverter = {
         return arr.join("&");
     },
 
+    sanitizeArgs: function(curlString) {
+        //replace -XPOST with -X POST
+        curlString = curlString.replace(/(-X)([A-Z]+)/, function (match, x, method) {return x + " " + method;})
+
+        var argv = shellQuote.parse("node " + curlString, function(key) {
+            // this is done to prevent converting vars like $id in the curl input to ''
+            return '$' + key;
+        });
+        var sanitizedArgs = _.map(_.filter(argv, function(arg) { return !_.isEmpty(arg) }), function (arg) {
+          if (_.isObject(arg) && arg.op === 'glob') {
+            return arg.pattern
+          }
+          else if (arg.op && arg.op.startsWith('$') && arg.op.length > 3) {
+            // in the app, certain headers like -H $'cookie: abc' are treated as operators
+            // converting the arg to cookie: abc instead of op: $'cookie: abc'
+            return arg.op.substring(2, arg.op.length-1);
+          }
+          else if (arg.startsWith('$') && arg.length > 2) {
+            // removing $ before every string like -X $'POST' and
+            // converting the arg to -X 'POST'
+            // link of RFC- https://tiswww.case.edu/php/chet/bash/bashref.html#ANSI_002dC-Quoting
```

Comment:
This is technically not an RFC. See if you can find a more authoritative doc. Check https://unix.stackexchange.com/questions/48106/what-does-it-mean-to-have-a-dollarsign-prefixed-string-in-a-script. This link should be included in the comments too.

Reply:
But both links have the same doc.